pf/internal/schemashim/attr_schema.go

```diff
@@ -91,6 +91,10 @@ func (s *attrSchema) Elem() interface{} {
 		var res shim.Resource = newObjectPseudoResource(tt, s.attr.Nested())
 		return res
 	}
+	if tt, ok := t.(pfattr.TypeWithElementTypes); ok {
```

Comment:
Also here worried this will catch more than tuples..

Reply:
It shouldn't. See the [documentation](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/attr#TypeWithElementTypes) for `pfattr.TypeWithElementTypes`.

It says
> TypeWithElementTypes extends the Type interface to include information about the types of each element. **Element types are part of the definition of a tuple type**.

(emphasis mine)